### PR TITLE
Pull request to review all code

### PR DESCRIPTION
**Pull request by bsaund**

### CODE REVIEW

File: dual_arm_teleop/.idea/codeStyles/Project.xml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<component name="ProjectCodeStyleConfiguration">
```
Comment (bsaund) [inline dual_arm_teleop/.idea/codeStyles/Project.xml]: This file should not be in the git repo

File: dual_arm_teleop/.idea/codeStyles/codeStyleConfig.xml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<component name="ProjectCodeStyleConfiguration">
```
Comment (bsaund) [inline dual_arm_teleop/.idea/codeStyles/codeStyleConfig.xml]: This file should not be in the git repo

File: dual_arm_teleop/.idea/dual_arm_teleop.iml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (bsaund) [inline dual_arm_teleop/.idea/dual_arm_teleop.iml]: All .idea files should not be in the git repo

File: openvr_ros_driver/src/openvr_driver.cpp [hunk 4]
```diff
@@ -0,0 +1,360 @@
+#include "openvr_driver.h"
+
+// Destructor
+OpenVRDriver::~OpenVRDriver() {
+  if (m_pHMD != NULL)
+  {
+    vr::VR_Shutdown();
+    m_pHMD = NULL;
```
Comment (bsaund) [inline openvr_ros_driver/src/openvr_driver.cpp]: Unnecessary line?

File: openvr_ros_driver/src/openvr_driver.cpp [hunk 5]
```diff
@@ -0,0 +1,360 @@
...
+  {
+    // if not connected just skip the rest of the routine
+    if (!m_pHMD->IsTrackedDeviceConnected(unDevice))
+      continue;
+
+    vr::TrackedDevicePose_t trackedDevicePose;
+    vr::TrackedDevicePose_t *devicePose = &trackedDevicePose;
+
+    vr::VRControllerState_t controllerState;
+    vr::VRControllerState_t *ontrollerState_ptr = &controllerState;
```
Comment (bsaund) [inline openvr_ros_driver/src/openvr_driver.cpp]: Unused, also misspelled

File: openvr_ros_driver/src/openvr_driver.cpp [hunk 6]
```diff
@@ -0,0 +1,360 @@
...
+
+  // Process SteamVR device states
+  for (vr::TrackedDeviceIndex_t unDevice = 0; unDevice < vr::k_unMaxTrackedDeviceCount; unDevice++)
+  {
+    // if not connected just skip the rest of the routine
+    if (!m_pHMD->IsTrackedDeviceConnected(unDevice))
+      continue;
+
+    vr::TrackedDevicePose_t trackedDevicePose;
+    vr::TrackedDevicePose_t *devicePose = &trackedDevicePose;
```
Comment (bsaund) [inline openvr_ros_driver/src/openvr_driver.cpp]: No need to use both the pointer and object. Further comments will adjust code to just use object `trackedDevicePose`, not the pointer

File: openvr_ros_driver/src/openvr_driver.cpp [hunk 7]
```diff
@@ -0,0 +1,360 @@
...
+
+    vr::VRControllerState_t controllerState;
+    vr::VRControllerState_t *ontrollerState_ptr = &controllerState;
+
+    vr::HmdVector3_t position;
+    vr::HmdQuaternion_t quaternion;
+
+    bool bPoseValid = trackedDevicePose.bPoseIsValid;
```
Comment (bsaund) [inline openvr_ros_driver/src/openvr_driver.cpp]: All 4 of these are unused. Delete

File: openvr_ros_driver/src/openvr_driver.cpp [hunk 8]
```diff
@@ -0,0 +1,360 @@
...
+    vr::ETrackingResult eTrackingResult;
+
+    // Get what type of device it is and work with its data
+    vr::ETrackedDeviceClass trackedDeviceClass = vr::VRSystem()->GetTrackedDeviceClass(unDevice);
+    switch (trackedDeviceClass) {
+      case vr::ETrackedDeviceClass::TrackedDeviceClass_HMD:
+      {
+        vr::VRSystem()->GetDeviceToAbsoluteTrackingPose(vr::TrackingUniverseStanding, 0, &trackedDevicePose, 1);
+
+        position = getPosition(devicePose->mDeviceToAbsoluteTracking);
```
Comment (bsaund) [inline openvr_ros_driver/src/openvr_driver.cpp]: `trackedDevicePose.`...

File: openvr_ros_driver/src/openvr_driver.cpp [hunk 9]
```diff
@@ -0,0 +1,360 @@
...
+        quaternion = getRotation(devicePose->mDeviceToAbsoluteTracking);
+
+        vVel = trackedDevicePose.vVelocity;
```
Comment (bsaund) [inline openvr_ros_driver/src/openvr_driver.cpp]: Unused. Remove these 4 lines.

File: openvr_ros_driver/src/openvr_driver.cpp [hunk 10]
```diff
@@ -0,0 +1,360 @@
...
+      }
+
+      case vr::ETrackedDeviceClass::TrackedDeviceClass_Controller:
+      {
+        vr::VRSystem()->GetControllerStateWithPose(vr::TrackingUniverseStanding, unDevice, &controllerState, sizeof(controllerState), &trackedDevicePose);
+
+        position = getPosition(devicePose->mDeviceToAbsoluteTracking);
+        quaternion = getRotation(devicePose->mDeviceToAbsoluteTracking);
+
+        vVel = trackedDevicePose.vVelocity;
```
Comment (bsaund) [inline openvr_ros_driver/src/openvr_driver.cpp]: Unused. Remove these 4 lines.

File: openvr_ros_driver/src/openvr_driver.cpp [hunk 11]
```diff
@@ -0,0 +1,360 @@
...
+        msg.hmd = msg_hmd;
+
+        break;
+      }
+
+      case vr::ETrackedDeviceClass::TrackedDeviceClass_Controller:
+      {
+        vr::VRSystem()->GetControllerStateWithPose(vr::TrackingUniverseStanding, unDevice, &controllerState, sizeof(controllerState), &trackedDevicePose);
+
+        position = getPosition(devicePose->mDeviceToAbsoluteTracking);
```
Comment (bsaund) [inline openvr_ros_driver/src/openvr_driver.cpp]: `trackedDevicePose.mDevice`...

File: openvr_ros_driver/src/openvr_driver.cpp [hunk 12]
```diff
@@ -0,0 +1,360 @@
...
+        break;
+      }
+
+      case vr::ETrackedDeviceClass::TrackedDeviceClass_TrackingReference:
+        vr::VRSystem()->GetDeviceToAbsoluteTrackingPose(vr::TrackingUniverseStanding, 0, &trackedDevicePose, 1);
+
+        position = getPosition(devicePose->mDeviceToAbsoluteTracking);
+        quaternion = getRotation(devicePose->mDeviceToAbsoluteTracking);
+
+        vVel = trackedDevicePose.vVelocity;
```
Comment (bsaund) [inline openvr_ros_driver/src/openvr_driver.cpp]: Unused, delete these 4 lines

File: openvr_ros_driver/src/openvr_driver.cpp [hunk 13]
```diff
@@ -0,0 +1,360 @@
...
+        // Header
+        msg_controller.posestamped.header.stamp = ros::Time::now();
+        msg_controller.posestamped.header.frame_id = "vive_base";
+
+        msg.controllers.push_back(msg_controller);
+
+        break;
+      }
+
+      case vr::ETrackedDeviceClass::TrackedDeviceClass_TrackingReference:
```
Comment (bsaund) [inline openvr_ros_driver/src/openvr_driver.cpp]: This case does not do anything, but we might want to expand this in the future. Add a comment like `//Currently tracking reference is not used`

File: openvr_ros_driver/src/openvr_driver.cpp [hunk 14]
```diff
@@ -0,0 +1,360 @@
...
+        position = getPosition(devicePose->mDeviceToAbsoluteTracking);
+        quaternion = getRotation(devicePose->mDeviceToAbsoluteTracking);
+
+        vVel = trackedDevicePose.vVelocity;
+        vAngVel = trackedDevicePose.vAngularVelocity;
+        eTrackingResult = trackedDevicePose.eTrackingResult;
+        bPoseValid = trackedDevicePose.bPoseIsValid;
+
+        break;
+    }
```
Comment (bsaund) [inline openvr_ros_driver/src/openvr_driver.cpp]: Add a `default` switch statement

File: vive_msgs/msg/ViveSystem.msg [hunk 15]
```diff
@@ -0,0 +1,2 @@
+Controller[] controllers
```
Comment (bsaund) [inline vive_msgs/msg/ViveSystem.msg]: I suggest calling this file `Trackables.msg` or something.
It is already a `vive_msgs` so `Vive` saying again is redundant.

Comment (andriyasdf) [reply vive_msgs/msg/ViveSystem.msg]: Perhaps `vive_msgs` should be renamed to `openvr_msgs`, as nothing in the message type is vive-specific.

**Pull request closed without merging**